robottelo/utils/virtwho.py

```diff
@@ -87,31 +88,52 @@ def runcmd(cmd, system=None, timeout=600000, output_format='base'):
     return result.status, result.stdout.strip()
 
 
-def register_system(system, activation_key=None, org='Default_Organization', env='Library'):
+def register_system(
+    system, activation_key=None, org='Default_Organization', env='Library', target=None
+):
     """Return True if the system is registered to satellite successfully.
 
     :param dict system: system account used by ssh to connect and register.
     :param str activation_key: the activation key will be used to register.
-    :param str org: Which organization will be used to register.
+    :param str org: Which organization will be used to register (org label).
     :param str env: Which environment will be used to register.
+    :param target: Satellite object for registration.
     :raises: VirtWhoError: If failed to register the system.
     """
-    runcmd('subscription-manager unregister', system)
-    runcmd('subscription-manager clean', system)
-    runcmd('rpm -qa | grep katello-ca-consumer | xargs rpm -e |sort', system)
-    runcmd(
-        f'rpm -ihv http://{settings.server.hostname}/pub/katello-ca-consumer-latest.noarch.rpm',
-        system,
+    # Get the guest hostname to check if it's already registered
+    guest_name, _ = runcmd('hostname', system=system)
+
+    if existing_hosts := Host.list({'search': guest_name}):
+        for host in existing_hosts:
+            Host.delete({'id': host['id']})
+
+    # Create ContentHost object from system dict
+    contenthost = ContentHost(
+        hostname=system['hostname'], auth=(system['username'], system['password'])
```

Comment:
For this case system username and password would be default settings.server.ssh_username and settings.server.ssh_password, which should be already set for ContentHost object